src/main/java/hudson/plugins/parameterizedtrigger/BuildTrigger.java

```diff
@@ -50,18 +47,39 @@ public BuildStepMonitor getRequiredMonitorService() {
 	@Override @SuppressWarnings("deprecation")
 	public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
 			BuildListener listener) throws InterruptedException, IOException {
+
+		HashSet<BuildTriggerConfig> alreadyFired = new HashSet<BuildTriggerConfig>();
+
+		// If this project has non-abstract projects, we need to fire them
+		for (BuildTriggerConfig config : configs) {
+			boolean hasNonAbstractProject = false;
+
+			List<Job> jobs = config.getProjectList(build.getRootBuild().getProject().getParent(), build.getEnvironment(listener));
+			for (Job j : jobs) {
+				if (!(j instanceof AbstractProject)) {
+					hasNonAbstractProject = true;
+				}
+			}
+			// Fire this config's projects if not already fired
+			if (hasNonAbstractProject) {
+				config.perform(build, launcher, listener);
+				alreadyFired.add(config);
+			}
+		}
+
         if (canDeclare(build.getProject())) {
             // job will get triggered by dependency graph, so we have to capture buildEnvironment NOW before
             // hudson.model.AbstractBuild.AbstractBuildExecution#cleanUp is called and reset
             EnvVars env = build.getEnvironment(listener);
             build.addAction(new CapturedEnvironmentAction(env));
-        } else {
+        } else {  // Not using dependency graph
             for (BuildTriggerConfig config : configs) {
-                config.perform(build, launcher, listener);
+				if (!alreadyFired.contains(config)) {
```

Comment:
so `canDeclare` will always be false for workflow (as it doesn't support the dependencyGraph), so the `else` will always be entered.  So I'm still confused.